subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_lte_notification.c

```diff
@@ -94,9 +98,57 @@ int lwm2m_update_signal_meas_objects(const struct lte_lc_cells_info *const cells
 	return 0;
 }
 
-void ncell_notification_handler(const struct lte_lc_evt *const evt)
+
+
+int lwm2m_ncell_handler_register(void)
+{
+	LOG_INF("Registering ncell notification handler");
+	if (!event_handler_requested) {
+		event_handler_requested = true;
+		lte_lc_register_handler(lte_notify_handler);
+	}
+	
+	return 0;
+}
+#endif
+
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_NWK_REG_NOTIFICATION)
+static lwm2m_lte_nwk_reg_update_state_cb_t lte_nwk_reg_cb;
+
+int lwm2m_lte_reg_handler_register(lwm2m_lte_nwk_reg_update_state_cb_t cb)
+{
+if (!event_handler_requested) {
+		event_handler_requested = true;
+		lte_lc_register_handler(lte_notify_handler);
```

Comment:
If you would create a common init instead, you wouldn't need register protection boolean. Have another function to set the callback.

Reply:
OK I create a common init.